lib/matplotlib/tests/test_backend_bases.py

```diff
@@ -349,3 +350,53 @@ def test_interactive_pan(key, mouseend, expectedxlim, expectedylim):
     # Should be close, but won't be exact due to screen integer resolution
     assert tuple(ax.get_xlim()) == pytest.approx(expectedxlim, abs=0.02)
     assert tuple(ax.get_ylim()) == pytest.approx(expectedylim, abs=0.02)
+
+
+def test_toolmanager_remove():
+    expected_warning_regex = (
+        r"Treat the new Tool classes introduced in "
+        r"v[0-9]*.[0-9]* as experimental for now; "
+        "the API and rcParam may change in future versions.")
+    with pytest.warns(UserWarning, match=expected_warning_regex):
+        plt.rcParams['toolbar'] = 'toolmanager'
+    fig = plt.gcf()
+    initial_len = len(fig.canvas.manager.toolmanager.tools)
+    assert 'forward' in fig.canvas.manager.toolmanager.tools.keys()
+    fig.canvas.manager.toolmanager.remove_tool('forward')
+    assert len(fig.canvas.manager.toolmanager.tools) == initial_len - 1
+    assert 'forward' not in fig.canvas.manager.toolmanager.tools.keys()
+
+
+def test_toolmanager_get_tool():
+    expected_warning_regex = (
+        r"Treat the new Tool classes introduced in "
+        r"v[0-9]*.[0-9]* as experimental for now; "
+        "the API and rcParam may change in future versions.")
+    with pytest.warns(UserWarning, match=expected_warning_regex):
+        plt.rcParams['toolbar'] = 'toolmanager'
+    fig = plt.gcf()
+    rubberband = fig.canvas.manager.toolmanager.get_tool('rubberband')
+    assert isinstance(rubberband, RubberbandBase)
+    assert fig.canvas.manager.toolmanager.get_tool(rubberband) == rubberband
```

Comment:
Let's check identity here
```suggestion
    assert fig.canvas.manager.toolmanager.get_tool(rubberband) is rubberband
```